docs/Diagram-Guidelines.md

```diff
@@ -0,0 +1,45 @@
+# Diagram Guidelines
+
+## Preferred file format
+
+For complex diagrams, use the `.drawio.svg` format.
+
+Files with the `.drawio.svg` extension are SVG files with embedded [draw.io](https://www.diagrams.net/) source code. Using that format lends itself to a developer-friendly workflow: it is valid SVG, plays well with `git diff` and can be edited in lock-step using various online and offline flavours of draw.&#8203;io. If you use VS Code, you can use an [extension](https://marketplace.visualstudio.com/items?itemName=hediet.vscode-drawio) for draw.&#8203;io integration.
+
+Files in the `.drawio.svg` format can be processed offline.
+
+## Embedding a diagram into Markdown
+
+To embed a `.drawio.svg` file into Markdown, use the same syntax as for any image. Example: `![My diagram](my-diagram.drawio.svg)`
+
+Mind that GitHub doesn’t allow styling in Markdown documents. Where styling is allowed (e. g. in the exported brew.sh version of the documentation), always set a background colour of `white` for the diagram. That’s the colour draw.&#8203;io assumes, and keeps the diagram easy to read in dark mode without further customization. You can use the CSS selector `img[src$=".drawio.svg"]` for styling.
```

Comment:
```suggestion
Mind that GitHub doesn’t allow styling in Markdown documents. Where styling is allowed (e.g. in the exported brew.sh version of the documentation), always set a background colour of `white` for the diagram. That’s the colour draw.io assumes, and keeps the diagram easy to read in dark mode without further customization. You can use the CSS selector `img[src$=".drawio.svg"]` for styling.
```

Reply:
@MikeMcQuaid The `&#8203;` was deliberate. It keeps Markdown from converting the name to a clickable link, which adds up and distracts from the content.

Reply:
Given it's a valid/correct link and this will be edited in plain-text: I think it's more readable without.

Reply:
Fair enough!